src/wms/src/test/java/org/geoserver/wms/wms_1_3/CapabilitiesIntegrationTest.java

```diff
@@ -945,6 +946,181 @@ public void testLayerGroupsPerWorkspace() throws Exception {
         }
     }
 
+    @Test
+    public void testRootLayerRemovedWMSService() throws Exception {
+        // make layers non advertised
+        WMSInfo info = getWMS().getServiceInfo();
+        info.getMetadata().put(WMS.NO_ROOT_LAYER_IN_CAPABILITIES_KEY, true);
+        getGeoServer().save(info);
+        Catalog catalog = getCatalog();
+        LayerGroupInfo group = null;
+        try {
+            group = createLakesPlacesLayerGroup(catalog, LayerGroupInfo.Mode.NAMED, null);
+            Document dom =
+                    getAsDOM(
+                            "lakes_and_places/wms?service=WMS&request=getCapabilities&version=1.3.0");
+            // print(dom);
+
+            assertXpathEvaluatesTo(
+                    "lakes_and_places",
+                    "/wms:WMS_Capabilities/wms:Capability/wms:Layer/wms:Name",
+                    dom);
+        } finally {
+            if (group != null) {
+                catalog.remove(group);
+            }
+            info.getMetadata().remove(WMS.NO_ROOT_LAYER_IN_CAPABILITIES_KEY);
+        }
+    }
+
+    @Test
+    public void testRootLayerNotRemoved() throws Exception {
+        // make layers non advertised
+        WMSInfo info = getWMS().getServiceInfo();
+        info.getMetadata().put(WMS.NO_ROOT_LAYER_IN_CAPABILITIES_KEY, false);
+        getGeoServer().save(info);
+        Catalog catalog = getCatalog();
+        LayerGroupInfo group = null;
+        try {
+            group = createLakesPlacesLayerGroup(catalog, LayerGroupInfo.Mode.NAMED, null);
+            Document dom = getAsDOM("lakes_and_places/wms?request=GetCapabilities&version=1.3.0");
+            // print(dom);
+
+            assertXpathEvaluatesTo(
+                    "", "/wms:WMS_Capabilities/wms:Capability/wms:Layer/wms:Name", dom);
+            assertXpathEvaluatesTo(
+                    "lakes_and_places",
+                    "/wms:WMS_Capabilities/wms:Capability/wms:Layer/wms:Layer/wms:Name",
+                    dom);
+        } finally {
+            if (group != null) {
+                catalog.remove(group);
+            }
+            info.getMetadata().remove(WMS.NO_ROOT_LAYER_IN_CAPABILITIES_KEY);
+        }
+    }
+
+    @Test
+    public void testRootLayerRemovedGroupConfig() throws Exception {
+        // make layers non advertised
```

Comment:
Actually it does not make them non advertised and/or not queriable... maybe this version of the tests (1.3 vs 1.1) got rid of the unecessary bits? These tests are much easier to read!

Reply:
Also, maybe it's interesting to see a test where a bunch of layers but one got marked as non advertised (hence not visible in the caps) and verify that the lone remaining layer got moved to the top

Reply:
Yes, I am going to add that (and verify the 1.1.1 behaviour related to advertising)